crates/vm/levm/src/precompiles.rs

```diff
@@ -232,12 +236,105 @@ pub fn ripemd_160(
+    consumed_gas: &mut U256,
 ) -> Result<Bytes, VMError> {
-    Ok(Bytes::new())
+    // If calldata does not reach the required length, we should fill the rest with zeros
+    let calldata = fill_with_zeros(calldata, 96)?;
+
+    let b_size: U256 = calldata
+        .get(0..32)
+        .ok_or(PrecompileError::ParsingInputError)?
+        .into();
+
+    let e_size: U256 = calldata
+        .get(32..64)
+        .ok_or(PrecompileError::ParsingInputError)?
+        .into();
+
+    let m_size: U256 = calldata
+        .get(64..96)
+        .ok_or(PrecompileError::ParsingInputError)?
+        .into();
+
+    if b_size == U256::zero() && m_size == U256::zero() {
+        *consumed_gas = consumed_gas
+            .checked_add(U256::from(MODEXP_STATIC_COST))
+            .ok_or(OutOfGasError::ConsumedGasOverflow)?;
+        return Ok(Bytes::new());
+    }
+
+    // Because on some cases conversions exploded before the if above
+    let b_size = usize::try_from(b_size).map_err(|_| PrecompileError::ParsingInputError)?;
+    let e_size = usize::try_from(e_size).map_err(|_| PrecompileError::ParsingInputError)?;
+    let m_size = usize::try_from(m_size).map_err(|_| PrecompileError::ParsingInputError)?;
+
+    let base_limit = b_size
+        .checked_add(96)
+        .ok_or(InternalError::ArithmeticOperationOverflow)?;
+
+    let exponent_limit = e_size
+        .checked_add(base_limit)
+        .ok_or(InternalError::ArithmeticOperationOverflow)?;
+
+    // The reason I use unwrap_or_default is to cover the case where calldata does not reach the required
+    // length, so then we should fill the rest with zeros. The same is done in modulus parsing
+    let b = calldata.get(96..base_limit).unwrap_or_default();
+    let base = BigUint::from_bytes_be(b);
+
+    let e = calldata.get(base_limit..exponent_limit).unwrap_or_default();
+    let exponent = BigUint::from_bytes_be(e);
+
+    let m = match calldata.get(exponent_limit..) {
+        Some(m) => {
+            let m_extended = fill_with_zeros(&Bytes::from(m.to_vec()), m_size)?;
+            m_extended.get(..m_size).unwrap_or_default().to_vec()
+        }
+        None => Default::default(),
+    };
+    let modulus = BigUint::from_bytes_be(&m);
+
+    let gas_cost = modexp_cost(&exponent, b_size, e_size, m_size)?;
+    increase_precompile_consumed_gas(gas_for_call, gas_cost, consumed_gas)?;
+
+    let result = mod_exp(base, exponent, modulus);
+
+    let res_bytes = result.to_bytes_be();
+    let res_bytes = increase_left_pad(&Bytes::from(res_bytes), m_size)?;
+
+    Ok(res_bytes.slice(..m_size))
+}
+
+/// I allow this clippy alert because in the code modulus could never be
+///  zero because that case is covered in the if above that line
+#[allow(clippy::arithmetic_side_effects)]
+fn mod_exp(base: BigUint, exponent: BigUint, modulus: BigUint) -> BigUint {
+    if modulus == BigUint::ZERO {
+        BigUint::ZERO
+    } else if exponent == BigUint::ZERO {
+        BigUint::from(1_u8) % modulus
```

Comment:
There is no "checked_rem" implemented for BigUint. To avoid the allow I'd have to turn it into a common type, do the operation, and then go back to BigUint.